skbio/io/format/genbank.py

```diff
@@ -0,0 +1,468 @@
+
+Feature Table Documentation:
+http://www.insdc.org/files/feature_table.html
+ftp://ftp.ncbi.nih.gov/genbank/docs/FTv10_3.html
+
+    The International Nucleotide Sequence Database Collaboration (INSDC)
+    between the DDBJ, EMBL, and GenBank.  These organisations all use the
+    same "Feature Table" layout in their plain text flat file formats.
+
+    However, the header and sequence sections of an EMBL file are very
+    different in layout to those produced by GenBank/DDBJ.
+
+Examples
+--------
+
+"""
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import range, zip
+from six.moves import zip_longest
+import re
+import textwrap
+
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from functools import partial
+
+from skbio.io import create_format, GenbankFormatError
+from skbio.io.registry import FileSentinel
+from skbio.io.format._base import (_get_nth_sequence,
+                                   _parse_fasta_like_header,
+                                   _format_fasta_like_records, _line_generator,
+                                   _too_many_blanks)
+from skbio.util._misc import chunk_str
+from skbio.sequence import Sequence, DNA, RNA, Protein
+from pprint import pprint
+
+genbank = create_format('genbank')
+
+# date format in locus line of genbank record
+_TIME_FORMAT = '%d-%b-%Y'
+# This list is ordered
+# used to read and write genbank file.
+_HEADERS = ['LOCUS',
+            'DEFINITION',
+            'ACCESSION',
+            'VERSION',
+            'DBSOURCE',
+            'DBLINK',
+            'KEYWORDS',
+            'SOURCE',
+            'REFERENCE',
+            'COMMENT',
+            'FEATURES',
+            'ORIGIN']
+
+
+@genbank.sniffer()
+def _genbank_sniffer(fh):
+    # check the 1st line starts with 'LOCUS' and the last line ends with '//'
+    if _too_many_blanks(fh, 5):
+        return False, {}
+    try:
+        line = next(_line_generator(fh, skip_blanks=True))
+    except StopIteration:
+        return False, {}
+
+    if line.startswith(_HEADERS[0]):
+        for line in _line_generator(fh, skip_blanks=True):
+            if line == '//':
+                return True, {}
+        return False, {}
+    else:
+        return False, {}
+
+
+@genbank.reader(None)
+def _genbank_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_genbanks(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@genbank.reader(Sequence)
+def _genbank_to_biological_sequence(fh, rec_num=1, **kwargs):
```

Comment:
I'm a little torn on this one, should `rec_num` be `seq_num` to match the other formats? I can see you are matching the style, and these are records, but scikit-bio will see them as sequences. @jairideout thoughts?